src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double baseLeg;
+    private double topLeg;
+    private double sideLeg;
+    private double height;
+
+    public IsoscelesTrapezoid(String color, double baseLeg, double topLeg, double sideLeg) {
+        super(color);
+        this.baseLeg = baseLeg;
+        this.topLeg = topLeg;
+        this.sideLeg = sideLeg;
+    }
+
+    @Override
+    public double calculateArea() {
+        height = Math.sqrt((4 * Math.pow(sideLeg, 2))
+                - Math.pow((topLeg > baseLeg ? topLeg - baseLeg : baseLeg - topLeg), 2));
+        return ((topLeg + baseLeg) * height) / 2;
+    }
+
+    @Override
+    public String display() {
+        return "Figure: right rectangle, area " + String.format("%.2f", calculateArea())
+                + " sq. units, baseLeg " + baseLeg + " units, topLeg " + topLeg
+                + " units, sideLeg " + sideLeg + " units, color: " + getColor();

```

Comment:
There is a bad solution for String concatenation with + operator. Read [this](https://github.com/mate-academy/jv-oop-advanced/pull/1273/javatpoint.com/java-string-format) article please.